Client/control/actions.py

```
@@ -0,0 +1,54 @@
+            return {"status": "error", "message": f"Invalid request: {error_msg}"}
+
+        action_name = action_request.get("action")
+        params = action_request.get("params", {})
+
+        if action_name not in self.actions:
+            return {"status": "error", "message": f"Unknown action: {action_name}"}
+
+        try:
+            func = self.actions[action_name]
+            result = func(**params)
+            return {"status": "success", "result": result}
```

Ruff (0.15.0)

[warning] Consider moving this statement to an `else` block

(TRY300)